controllers/gateway/api_controller_integration_test.go

```diff
@@ -43,6 +44,127 @@ var _ = Describe("APIRule Controller", Serial, func() {
 		defaultHttpTimeout         = time.Second * 180
 	)
 
+	Context("check default domain logic", func() {
+
+		It("should have an error when creating an APIRule without a domain in cluster without kyma-gateway", func() {
+			updateJwtHandlerTo(helpers.JWT_HANDLER_ORY)
+
+			rule1 := testRule("/rule1", []string{"GET"}, defaultMutators, noConfigHandler("allow"))
+
+			apiRuleName := generateTestName(testNameBase, testIDLength)
+			serviceName := generateTestName(testServiceNameBase, testIDLength)
+			serviceHost := serviceName
+
+			By("Creating APIRule")
+
+			apiRule := testApiRule(apiRuleName, testNamespace, serviceName, testNamespace, serviceHost, testServicePort, []gatewayv1beta1.Rule{rule1})
+			svc := testService(serviceName, testNamespace, testServicePort)
+
+			// when
+			Expect(c.Create(context.TODO(), svc)).Should(Succeed())
+			Expect(c.Create(context.TODO(), apiRule)).Should(Succeed())
+			defer func() {
+				apiRuleTeardown(apiRule)
+				serviceTeardown(svc)
+			}()
+
+			expectApiRuleStatus(apiRuleName, gatewayv1beta1.StatusError)
```

Comment:
Shouldn't the test end here? I was confused that there is an APIRule update after that, since the test case mentions the error as the verification step.Do we have details in the error description that can be user for further validation, so we can remove the rest of the code in this test?

Reply:
I wanted to additionally test that supplying a full domain `httpbin.local.kyma.dev` will fix the error on APIRule
If you feel like it would be better to remove that, I can do it

Reply:
In my opinion if we want to do it like that, then we should adjust the test description, so it's clear that the error is not the final state of the test.

Reply:
15a0087...56d2aaa